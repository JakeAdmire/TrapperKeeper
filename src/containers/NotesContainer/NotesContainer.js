import React, { Component } from 'react';
import { Link} from 'react-router-dom';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types'

export class NotesContainer extends Component {

  generateNotes = () => {
    return this.props.allNotes.map(note =>
      <Link to={`/notes/${note.id}`} key={note.id} className='note-click'>
        <div>
          <h3>{note.title}</h3>
        </div>
      </Link>
    )
  }

  render() {
    let splashMessage = this.props.allNotes.length ? null : <h1 className="splash-message">Try Creating a New Note!</h1>
    return (
      <div className="NotesContainer">
        { splashMessage }
        { this.props.allNotes && this.generateNotes() }
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  allNotes: state.allNotes,
})

NotesContainer.propTypes = {
  allNotes: PropTypes.array
}

export default connect(mapStateToProps)(NotesContainer);