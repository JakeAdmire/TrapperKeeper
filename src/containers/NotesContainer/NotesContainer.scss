.NotesContainer {
  display: flex;
  flex-wrap: wrap;

  a {
    width: 23%;
    min-height: 100px;
    margin: 1%;
    border-radius: 15px;
    border: solid hsl(0, 0%, 88%) 1px;
    background: white;
    text-decoration: none;
    padding: 1em;
    display: flex;
    font-family: $font-Abel;
    box-shadow: 0 0 6px 3px #00000014;
  }

  h3 {
    color: #e0e0e0;
    margin: 0;
  }

  .splash-message {
    margin: 100px auto 0;
    color: #75757580;
    font-size: 80px;
    width: 500px;
  }
  
}
