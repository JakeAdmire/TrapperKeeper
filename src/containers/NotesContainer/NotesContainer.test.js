import React from 'react';
import { shallow } from 'enzyme';
import { NotesContainer } from './NotesContainer';


describe('NotesContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<NotesContainer />)
  })

  it('should match the snapshot', () => {

  })

})
