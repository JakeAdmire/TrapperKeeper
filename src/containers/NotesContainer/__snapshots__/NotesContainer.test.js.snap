// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesContainer NotesContainer should match the snapshot 1`] = `
<div
  className="NotesContainer"
>
  <Link
    className="note-click"
    key="123ABA"
    to="/notes/123ABA"
  >
    <NoteDisplay
      id="123ABA"
      list={
        Array [
          Object {
            "id": [MockFunction],
            "isComplete": false,
            "text": "Eat food",
          },
        ]
      }
      title="Worf ToDo"
    />
  </Link>
  <Link
    className="note-click"
    key="987GDGFD"
    to="/notes/987GDGFD"
  >
    <NoteDisplay
      id="987GDGFD"
      list={
        Array [
          Object {
            "id": [MockFunction],
            "isComplete": false,
            "text": "Do basic styling",
          },
        ]
      }
      title="Jake ToDo"
    />
  </Link>
</div>
`;

exports[`NotesContainer NotesContainer should return JSX when generateNotes is called 1`] = `undefined`;
